components/layout/display_list/largest_contenful_paint_collector.rs

```diff
@@ -0,0 +1,174 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use base::id::ScrollTreeNodeId;
+use compositing_traits::display_list::{ScrollTree, SpatialTreeNodeInfo};
+use compositing_traits::largest_contentful_paint_candidate::{ImageCandidate, LCPCandidates};
+use log::debug;
+use webrender_api::Epoch;
+use webrender_api::units::{LayoutRect, LayoutSize};
+
+use crate::display_list::clip::{Clip, ClipId, StackingContextTreeClipStore};
+
+pub(crate) struct LargestContentfulPaintCollector<'a> {
+    pub clip_tree: &'a StackingContextTreeClipStore,
+
+    pub current_scroll_node_id: ScrollTreeNodeId,
+
+    pub current_clip_id: ClipId,
+
+    pub largerst_image: Option<ImageCandidate>,
+
+    pub lcp_candidates: &'a mut LCPCandidates,
+}
+
+impl LargestContentfulPaintCollector<'_> {
+    pub fn update_current_node_id(
+        &mut self,
+        next_scroll_node_id: ScrollTreeNodeId,
+        next_clip_id: ClipId,
+    ) {
+        self.current_scroll_node_id = next_scroll_node_id;
+        self.current_clip_id = next_clip_id;
+    }
+
+    pub fn record_image_candidate(
+        &mut self,
+        tag: usize,
+        rect: LayoutRect,
+        viewport_size: LayoutSize,
+        scroll_tree: &ScrollTree,
+        epoch: Epoch,
+    ) {
+        let visual_rect = self.compute_visual_rect(rect, scroll_tree, viewport_size);
+        let size = visual_rect.area() as usize;
```

Comment:
I would recommend renaming `size` to area, as that makes it more clear what we are measuring. 
That includes the name in `ImageCandidate`